sd: resend CMD0 before each CMD8

JordanYates opened the pull request:
Resend CMD0 before each CMD8 query while initialising the card, as the
first CMD0 is not always sufficient to recover the card.

This resolves an issue found while stress testing a hardware platform that has an SD card on a power domain and a flash chip on the same SPI bus. Only running CMD0 at the start was failing to put the card into a good state, while the updated implementation reliably works on the second go around.

decsny commented on subsys/sd/sd.c:
```diff
@@ -94,14 +94,8 @@ static int sd_common_init(struct sd_card *card)
 	int ret;
 
-	/* Reset card with CMD0 */
-	ret = sd_idle(card);
```
why can't you just use sd_retry to retry sending the cmd0?

JordanYates replied:
Because the whole purpose of CMD0 here is to make the card respond to CMD8. There is no way to tell if CMD0 has worked independently of another command

decsny replied:
I guess what I mean is like can you make a function that does CMD0 and CMD8 and call that with the sd_retry rather than changing this function signature for a specific case

JordanYates replied:
That makes much more sense, updated PR

JordanYates committed:
sd: resend CMD0 before each CMD8

Resend CMD0 before each CMD8 query while initialising the card, as the
first CMD0 is not always sufficient to recover the card.

Signed-off-by: Jordan Yates <jordan@embeint.com>
